src/aind_metadata_mapper/smartspim/acquisition.py

```diff
@@ -0,0 +1,271 @@
+"""SmartSPIM ETL to map metadata"""
+
+import os
+import re
+from datetime import datetime
+from pathlib import Path
+from typing import Dict
+
+from aind_data_schema.components.coordinates import ImageAxis
+from aind_data_schema.core import acquisition
+from pydantic_settings import BaseSettings
+
+from aind_metadata_mapper.core import GenericEtl, JobResponse
+
+from .utils import (
+    get_anatomical_direction,
+    get_excitation_emission_waves,
+    get_session_end,
+    make_acq_tiles,
+    read_json_as_dict,
+)
+
+
+class JobSettings(BaseSettings):
+    """Data to be entered by the user."""
+
+    subject_id: str
+    raw_dataset_path: Path
+    output_directory: Path
+
+    # Metadata names
+    asi_filename: str = "derivatives/ASI_logging.txt"
+    mdata_filename_json: str = "derivatives/metadata.json"
+
+    # Metadata provided by microscope operators
+    processing_manifest_path: str = "derivatives/processing_manifest.json"
+
+
+class SmartspimETL(GenericEtl):
+    """
+    This class contains the methods to write the metadata
+    for a SmartSPIM session
+    """
+
+    def __init__(self, job_settings: BaseSettings):
+        """
+        Constructor method
+
+        Parameters
+        ----------
+        job_settings: BaseSettings
+            Job settings for the SmartSPIM ETL
+
+        """
+        if isinstance(job_settings, str):
+            job_settings_model = JobSettings.model_validate_json(job_settings)
+
+        else:
+            job_settings_model = job_settings
+
+        self.regex_date = r"(20[0-9]{2})-([0-9]{2})-([0-9]{2})_([0-9]{2})-([0-9]{2})-([0-9]{2})"  # noqa: E501
```

Comment:
There are ways to split strings across multiple lines like:
```python
    self.regex_date = (
        r"(20[0-9]{2})-([0-9]{2})-([0-9]{2})_([0-9]{2})-([0-9]{2})-([0-9]{2})"
    )
```
or even:
```python
    self.regex_date = (
        r"(20[0-9]{2})-([0-9]{2})-([0-9]{2})_([0-9]{2})-"
        r"([0-9]{2})-([0-9]{2})"
    )
```
which is usually a bit nicer than tagging lines with `# noqa` too often